crates/networking/discv5/src/subnet.rs

```diff
@@ -0,0 +1,149 @@
+
+    pub fn enable_subnet(&mut self, subnet: Subnet) -> anyhow::Result<()> {
+        match subnet {
+            Subnet::Attestation(id) if id < 64 => {
+                let bits = self.attestation_bits.get_or_insert(BitVector::new());
+                bits.set(id as usize, true)
+                    .map_err(|_| anyhow!("Subnet ID out of bounds"))?;
+                Ok(())
+            }
+            Subnet::Attestation(_) => Ok(()),
+            Subnet::SyncCommittee(_) => unimplemented!("SyncCommittee support not yet implemented"),
+        }
+    }
+
+    pub fn disable_subnet(&mut self, subnet: Subnet) -> anyhow::Result<()> {
+        match subnet {
+            Subnet::Attestation(id) if id < 64 => {
+                if let Some(bits) = &mut self.attestation_bits {
+                    bits.set(id as usize, false)
+                        .map_err(|_| anyhow!("Subnet ID out of bounds"))?;
+                }
+                Ok(())
+            }
+            Subnet::Attestation(_) => Ok(()),
+            Subnet::SyncCommittee(_) => unimplemented!("SyncCommittee support not yet implemented"),
+        }
+    }
+
+    pub fn is_active(&self, subnet: Subnet) -> bool {
+        match subnet {
+            Subnet::Attestation(id) if id < 64 => self
+                .attestation_bits
+                .as_ref()
+                .is_some_and(|bits| bits.get(id as usize).unwrap_or(false)),
+            Subnet::Attestation(_) => false,
+            Subnet::SyncCommittee(_) => unimplemented!("SyncCommittee support not yet implemented"),
+        }
+    }
+}
+
+impl Encodable for Subnets {
+    fn encode(&self, out: &mut dyn BufMut) {
+        let ssz_bytes = self.as_ssz_bytes();
+        let bytes = Bytes::from(ssz_bytes);
+        bytes.encode(out);
+    }
+}
+
+impl Decodable for Subnets {
+    fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
+        let bytes = Bytes::decode(buf)?;
+        let subnets = Subnets::from_ssz_bytes(&bytes)
+            .map_err(|_| alloy_rlp::Error::Custom("Failed to decode SSZ subnets"))?;
+        Ok(subnets)
+    }
+}
+
+pub fn subnet_predicate(subnets: Vec<Subnet>) -> impl Fn(&Enr) -> bool + Send + Sync {
+    move |enr: &Enr| {
+        let value = if let Some(result) = enr.get_decodable::<Bytes>(ATTESTATION_BITFIELD_ENR_KEY) {
+            match result {
+                Ok(value) => value,
+                Err(_) => return false,
+            }
+        } else {
+            return false;
+        };
+        let subnets_state = match Subnets::from_ssz_bytes(&value) {
+            Ok(subnets) => subnets,
+            Err(_e) => return false,
+        };
+        let attestation_bits = match &subnets_state.attestation_bits {
+            Some(bits) => bits,
+            None => {
+                trace!(
+                    "Peer rejected: invalid or missing attnets field; peer_id: {}",
+                    enr.node_id()
+                );
+                return false;
+            }
+        };
```

Comment:
```suggestion
        let Some(attestation_bits) = &subnets_state.attestation_bits else {
                trace!(
                    "Peer rejected: invalid or missing attnets field; peer_id: {}",
                    enr.node_id()
                );
                return false;
        };
```

This can be simplified to